java/code/src/com/suse/manager/webui/services/SaltActionChainGeneratorService.java

```diff
@@ -199,19 +210,30 @@ private Optional<Long> nextActionId(List<SaltState> states, int currentPos) {
     }
 
     private SaltState endChunk(ActionChain actionChain, int chunk, Optional<Long> nextActionId,
-                               Optional<Pair<String, String>> lastRef, Optional<String> sshExtraFileRefs) {
+                               Optional<Pair<String, String>> lastRef, Optional<String> sshExtraFileRefs,
+                               Optional<Boolean> rebootRequired) {
         Map<String, Object> args = new LinkedHashMap<>(2);
         args.put("actionchain_id", actionChain.getId());
         args.put("chunk", chunk + 1);
         nextActionId.ifPresent(actionId ->
                 args.put("next_action_id", actionId));
         sshExtraFileRefs.ifPresent(refs ->
                 args.put("ssh_extra_filerefs", refs));
+        rebootRequired.ifPresent(refs ->
+                args.put("reboot_required", refs));
```

Comment:
if we do not check for "true" or "false", why do we need an Optional?
Could we just use "boolean" and set `True` and `False` ?
In this case it is impossible to not set this value.

Reply:
Since this parameter is  used just for transactional update minion, I thought it was misleading to have `reboot_required: False` for non-transactional update system (since it's never evaluated). But I don't have a strong opinion about that.
Probably the best would be always set `reboot_required` just transactional update system, and use `Optional.empty()` for traditional system , what do you think?

Reply:
In Line 179 you set reboot to `Optional.of(false)`. But this has here now the same effect as `Optional.of(true)`.
I think this is a bug.

Reply:
Ahh, sorry. Forget it. you set the value to the arg and not hard true.
This looks correct.

Reply:
I'm wondering if there's a better variable (and arg) name than `reboot_required`. I think it can be misleading in this context, since in other context it means that the user should run a reboot... but I didn't find anything better :)